File: fip_collab/strain_mks/matthew_250FZreducedNewBCs/analysis_elastic_strain_step06/euler_to_gsh.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jul 15 11:45:46 2014

@author: nhpnp3
"""

import numpy as np
import GSH_func as gsh
#import scipy.io as sio

ns = 50
el = 21
set_id = 'val'

euler = np.load('euler_%s%s.npy' %(ns,set_id))

euler_GSH = np.zeros([el**3,ns,15], dtype= 'complex128')

for sn in range(ns):
    for k in range(el**3):
        euler_GSH[k,sn,:] = gsh.GSH_Hexagonal_Triclinic(euler[k,sn,:])
    print sn

np.save('euler_GSH_%s%s.npy' %(ns,set_id),euler_GSH)


### Let us compare this result to the matlab result:
#
#euler_GSH_matlab = sio.loadmat('euler_GSH_%s%s.mat' %(ns,set_id))['euler_GSH']
#euler_GSH = np.load('euler_GSH_%s%s.npy' %(ns,set_id))
#
#print np.sum(np.absolute(euler_GSH_matlab - euler_GSH))




#euler = np.load('euler_%s%s.npy' %(ns,set_id))
#
#euler_GSH = np.zeros([el**3,ns,10], dtype= 'complex128')
#
#for sn in range(ns):
#    for k in range(el**3):
#        euler_GSH[k,sn,:] = gsh.GSH_Cubic_Triclinic(euler[k,sn,0],euler[k,sn,1],euler[k,sn,2])
#    print sn
#
#np.save('euler_GSH_%s%s.npy' %(ns,set_id),euler_GSH)
#
#
### Let us compare this result to the matlab result:
#
#euler_GSH_matlab = sio.loadmat('euler_cubic_GSH_%s%s.mat' %(ns,set_id))['euler_GSH']
#euler_GSH = np.load('euler_GSH_%s%s.npy' %(ns,set_id))
#
#print np.sum(np.absolute(euler_GSH_matlab - euler_GSH))

File: inp_generation/polycrystal_single_phase/Euler_to_GSH.m
ns = 200;
el = 21;

load euler_200cal.mat

euler_GSH = zeros(el^3,15,ns);

c = 0;

for sn = 1 : ns
    for k = 1 : el^3
        euler_GSH(k,:,sn)= GSH_Hexagonal_Triclinic(euler_200cal(k,sn,1),euler_200cal(k,sn,2),euler_200cal(k,sn,3));
        
        if any(isnan(euler_GSH(k,:,sn))) == 1
            c = c + 1;
            bad_euler(c,:) = euler_200cal(k,sn,:);
        end
        
    end
    sn
end

euler_GSH = permute(euler_GSH,[1 3 2]);

save('euler_GSH_200cal','euler_GSH');